usmqe_tests/api/others/test_user.py

```diff
@@ -0,0 +1,200 @@
+# -*- coding: utf8 -*-
+"""
+REST API test suite - user
+"""
+
+import json
+import pytest
+from usmqe.api.tendrlapi import user as tendrlapi_user
+from usmqe.api.tendrlapi.common import login, logout
+
+
+LOGGER = pytest.get_logger('user_test', module=True)
+"""@pylatest default
+Setup
+=====
+
+Prepare USM cluster accordingly to documentation.
+
+Further mentioned ``APIURL`` points to: ``http://USMSERVER/api/1.0``.
+"""
+
+"""@pylatest default
+Teardown
+========
+"""
+
+
+def test_user_get(valid_session_credentials, valid_new_user):
+    """@pylatest api/user.get
+    API-users: get user
+    *******************
+
+    .. test_metadata:: author mkudlej@redhat.com dahorak@redhat.com fbalak@redhat.com
+
+    Description
+    ===========
+
+    Get user from ``valid_new_user`` fixture.
+    """
+    test = tendrlapi_user.ApiUser(auth=valid_session_credentials)
+    """@pylatest api/user.get
+    .. test_step:: 2
+
+        Send **GET** request to ``APIURL/users``.
+
+    .. test_result:: 2
+
+        List of users in database is returned
+    """
+    test.users()
+    """@pylatest api/user.get
+    .. test_step:: 3
+
+        Get user info.
+
+        Send **GET** request to ``APIURL/users/{user}``.
+
+    .. test_result:: 3
+
+        User information for user from ``valid_new_user`` fixture is returned.
+    """
+    test.check_user(valid_new_user)
+
+
+def test_user_change_password(valid_new_user, valid_password):
+    """@pylatest api/user.edit
+    API-users: edit user
+    *******************
+
+    .. test_metadata:: author fbalak@redhat.com
+
+    Description
+    ===========
+
+    Change password and email of user and login with new password.
+    """
+    auth = login(valid_new_user["username"], valid_new_user["password"])
+    test = tendrlapi_user.ApiUser(auth=auth)
+    """@pylatest api/user.get
+    .. test_step:: 1
+
+        Send **PUT** request to ``APIURL/users``.
+
+    .. test_result:: 1
+
+        Edited user data are returned.
+    """
+    new_email = "testmail@tendrl.org"
+    edit_data = {
+        "email": new_email,
```

Comment:
This ensures that user have set up email. API call that changes password requires for user to have specified email. For example generated user admin doesn't have email set.